src/sidebar/store/create-store.js

```diff
@@ -83,6 +81,74 @@ import { createReducer, bindSelectors } from './util';
  *   SelectorMethods<RootSelectors>} Store
  */
 
+/**
+ * Create a Redux reducer from a store module's reducer map.
+ *
+ * @template State
+ * @param {ReducerMap<State>} reducers
+ */
+function createReducer(reducers) {
+  /** @param {redux.Action} action */
+  return (state = /** @type {State} */ ({}), action) => {
+    const reducer = reducers[action.type];
+    if (!reducer) {
+      return state;
+    }
+    const stateChanges = reducer(state, action);
+
+    // Some modules return an array rather than an object. They need to be
+    // handled differently so we don't convert them to an object.
+    if (Array.isArray(stateChanges)) {
+      return stateChanges;
+    }
+
+    return {
+      ...state,
+      ...stateChanges,
+    };
+  };
+}
+
+/**
+ * Convert a map of selector functions, which take a state value as their
+ * first argument, to a map of selector methods, which pre-fill the first
+ * argument by calling `getState()`.
+ *
+ * @template State
+ * @template {SelectorMap<State>} Selectors
+ * @param {Selectors} selectors
+ * @param {() => State} getState
+ * @return {SelectorMethods<Selectors>}
+ */
+function bindSelectors(selectors, getState) {
```

Comment:
I'm having a little trouble with plural vs. singular on some of these functions: `createReducer`, but `bindSelectors`. I have a sense an underlying redux construct may be at play here; can you enlighten me?

Reply:
`createReducer` is singular because the output is a single reducer function. `bindSelectors` takes an object whose property values are selector functions and returns an object with the same properties, but where the selectors have been modified to remove the first argument.